src/main/java/org/prebid/server/auction/PriceGranularity.java

```diff
@@ -25,70 +23,66 @@
+                range(5, 0.05),
+                range(10, 0.1),
+                range(20, 0.5));
+        putStringPriceGranularity(PriceGranularityType.dense, 2,
+                range(3, 0.01),
+                range(8, 0.05),
+                range(20, 0.5));
     }
 
     /**
      * Creates {@link PriceGranularity} for string representation and puts it to
      * {@link EnumMap<PriceGranularityType, PriceGranularity>}
      */
-    private static void putStringPriceGranularity(PriceGranularityType type, ExtPriceGranularityBucket... buckets) {
+    private static void putStringPriceGranularity(PriceGranularityType type, Integer presicion,
+                                                  ExtGranularityRange... ranges) {
         STRING_TO_CUSTOM_PRICE_GRANULARITY.put(type,
-                PriceGranularity.createFromBuckets(Arrays.asList(buckets)));
+                PriceGranularity.createFromRanges(presicion, Arrays.asList(ranges)));
     }
 
     /**
-     * Creates {@link ExtPriceGranularityBucket} from given precision, min, max and increment parameters.
+     * Creates {@link ExtGranularityRange`} from given precision, min, max and increment parameters.
      */
-    private static ExtPriceGranularityBucket bucket(int precision, int min, int max, double increment) {
-        return ExtPriceGranularityBucket.of(precision, BigDecimal.valueOf(min), BigDecimal.valueOf(max),
-                BigDecimal.valueOf(increment));
+    private static ExtGranularityRange range(int max, double increment) {
+        return ExtGranularityRange.of(BigDecimal.valueOf(max), BigDecimal.valueOf(increment));
     }
 
     public static final PriceGranularity DEFAULT = STRING_TO_CUSTOM_PRICE_GRANULARITY
             .get(PriceGranularityType.med);
 
-    private List<ExtPriceGranularityBucket> buckets;
-    private BigDecimal bucketMax;
+    private List<ExtGranularityRange> ranges;
+    private BigDecimal rangesMax;
     private Integer precision;
 
-    private PriceGranularity(List<ExtPriceGranularityBucket> buckets, BigDecimal bucketMax, Integer precision) {
-        this.buckets = buckets;
-        this.bucketMax = bucketMax;
+    private PriceGranularity(List<ExtGranularityRange> ranges, BigDecimal rangesMax, Integer precision) {
+        this.ranges = ranges;
+        this.rangesMax = rangesMax;
         this.precision = precision;
     }
 
     /**
-     * Creates {@link PriceGranularity} from {@link List<ExtPriceGranularityBucket>} and validates it.
+     * Creates {@link PriceGranularity} from {@link List<ExtGranularityRange>} and validates it.
      */
-    public static PriceGranularity createFromBuckets(List<ExtPriceGranularityBucket> buckets) {
-        if (CollectionUtils.isEmpty(buckets)) {
-            throw new IllegalArgumentException("Bucket list cannot be null or empty");
+    public static PriceGranularity createFromRanges(Integer precision, List<ExtGranularityRange> ranges) {
```

Comment:
This method could accept single `ExtPriceGranularity` object instead of two constituents of it.